packages/react/src/components/Swap/Swap.tsx

```diff
@@ -33,7 +54,51 @@ export const Swap = ({ tokenBondingKey }: { tokenBondingKey: PublicKey }) => {
     tradingMints,
     onTradingMintsChange: setTradingMints,
     swap: (args) =>
-      execute(args).then(({ targetAmount }) => {
+      execute({
+        async extraInstructions({isFirst, amount, isBuy}: IExtraInstructionArgs): Promise<InstructionResult<null>> {
+          if (!isFirst || !childEntangler || !parentEntangler || isBuy) {
+            return {
+              instructions: [],
+              signers: [],
+              output: null,
+            }
+          }
+
+          let numAmount = toNumber(amount!, childMint)
+          const instr = await fungibleEntanglerSdk?.swapParentForChildInstructions({
```

Comment:
Could probably just add fungibleEntanglerSdk to the if statement above and simplify this to a return.